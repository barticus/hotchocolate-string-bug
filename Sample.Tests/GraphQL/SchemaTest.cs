using System.Threading.Tasks;
using ApprovalTests;
using Microsoft.AspNetCore.TestHost;
using NUnit.Framework;

namespace Sample.Tests.GraphQL
{
    public class SchemaTest
    {
        public const string GraphqlPath = "graphql";

        protected TestServer Server { get; set; } = default!;

        [OneTimeSetUp]
        public void OneTimeSetUp()
        {
            Server = MockServer.Create();
        }

        [Test]
        public async Task GetSchema()
        {
            // Arrange
            var req = Server.CreateRequest(GraphqlPath + "/schema");

            // Act
            var result = await req.GetAsync();

            // Assert
            Assert.AreEqual(200, (int)result.StatusCode);
            Approvals.Verify(await result.Content.ReadAsStringAsync());
        }
    }
}
