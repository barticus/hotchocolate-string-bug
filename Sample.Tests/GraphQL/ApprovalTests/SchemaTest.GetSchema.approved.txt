schema {
  query: Query
}

type Comment {
  content: [String]!
  id: Uuid!
}

type Query {

}

"The `String` scalar type represents textual data, represented as UTF-8 character sequences. The String type is most often used by GraphQL to represent free-form human-readable text."
scalar String

scalar Uuid